src/core/utils.js

```diff
@@ -494,6 +495,30 @@ export function flatten(collector, item) {
 
 // --- DOM HELPERS -------------------------------
 
+// Takes an array and returns a string that separates each of its items with the proper commas and
+// "and". The second argument is a mapping function that can convert the items before they are
+// joined
+// Finally converts to hyperHTML while joining them
+
+export function htmlJoinAnd(array = [], mapper = item => item) {
+  const items = array.map(mapper);
+  switch (items.length) {
+    case 0:
+    case 1: // "x"
+      return items[0];
+    case 2: // x and y
+      return hyperHTML`${items[0]} and ${items[1]}`;
+    default: {
+      // x, y, and z
+      let joinedItems = `${items[0]}`;
+      for (let i = 1; i < items.length - 1; i++) {
+        joinedItems = hyperHTML`${joinedItems}, ${items[i]}`;
+      }
+      joinedItems = hyperHTML`${joinedItems} and ${items[length - 1]}`;
```

Comment:
* This still treats `items[0]` as a string 👀, we should also pass it into hyperHTML.
* Too easy to forget `hyperHTML` here, so how about:

   ```js
   const joinedItems = items.slice(-1).map(item => hyperHTML`${item}, `);
   return hyperHTML`${joinedItems} and ${items[items.length - 1]}`;
   ```

   IMO much shorter in this way. (The value of `joinedItems` never changes which leads to more readable code, and also adds a required comma before "and".)

* Note that `length` here is undefined, should be `items.length` instead.

Reply:
items is an array of hyperHTML elements. So items[0] should be an HTML element here.

I will just change length to items.length

Reply:
Also 
`const joinedItems = items.slice(-1).map(item => hyperHTML`${item}, `);
return hyperHTML'${joinedItems} and ${items[items.length - 1]}';`
will return the last element twice separated by both `,` and `and`
`const joinedItems = items.slice(0,-1).map(item => hyperHTML`${item}, `);
return hyperHTML'${joinedItems} and ${items[items.length - 1]}';`
will have the same problem as my previous code had.